platform/ios/src/MGLMapView.mm

```diff
@@ -4759,7 +4804,11 @@ - (void)cameraIsChanging {
     }
 
     [self updateCompass];
-
+    
+    if (!self.scaleBarView.hidden) {
+        [(MGLScaleBarView *)self.scaleBarView setMetersPerPoint:[self metersPerPointAtLatitude:self.centerCoordinate.latitude]];
```

Comment:
What if the scale bar is hidden while the user zooms the map, but then the developer unhides the scale bar afterwards? Would the scale bar remain accurate? Can we make it safe to call `-[MGLScaleBarView setMetersPerPoint:metersPerPointAtLatitude:]` even when MGLScaleBarView is hidden?